### PR TITLE
Bug 1068909 - improving accessibility for preview mode in Camera app.

### PR DESCRIPTION
**Pull request by yzen**

### CODE REVIEW

**Comment by try-server-hook** [comment]
[yzen Yura Zenevich (yzen)](https://github.com/yzen) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=807b81dc68e3)

**Comment by try-server-hook** [comment]
[yzen Yura Zenevich (yzen)](https://github.com/yzen) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=ff52498e4f7e)

**Comment by try-server-hook** [comment]
[yzen Yura Zenevich (yzen)](https://github.com/yzen) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=de045adad274)

**Comment by try-server-hook** [comment]
[yzen Yura Zenevich (yzen)](https://github.com/yzen) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=2915684964c5)

File: apps/camera/js/controllers/preview-gallery.js [hunk 1]
```diff
@@ -77,6 +77,11 @@ PreviewGalleryController.prototype.openPreview = function() {
   this.view.on('loadingvideo', this.app.firer('busy'));
   this.view.on('playingvideo', this.app.firer('ready'));
 
+  // Screen reader right/left swipes with two fingers to navigate to
+  // next/previous frames.
+  this.view.on('wheel:right', this.next);
+  this.view.on('wheel:left', this.previous);
```
Comment (justindarc) [inline apps/camera/js/controllers/preview-gallery.js]: I'm not familiar with this event. Is this something implemented by the screen reader?

Comment (yzen) [inline apps/camera/js/controllers/preview-gallery.js]: @justindarc , yes, screen reader synthesizes the wheel event when the screen reader user performs a swipe with 2 fingers.

File: apps/camera/js/views/controls.js [hunk 2]
```diff
@@ -245,11 +250,19 @@ module.exports = View.extend({
     this.unset(key ? key + '-enabled' : 'enabled');
   },
 
+  localize: function() {
+    for (var el in this.elsL10n) {
+      this.els[el].setAttribute('data-l10n-id', this.elsL10n[el]);
```
Comment (justindarc) [inline apps/camera/js/views/controls.js]: I'm not sure I understand the point of what this function is doing. Aren't the `data-l10n-id` attributes already hard-coded in the `template` below to the same values already defined in `this.elsL10n` above? I also don't see anywhere that `this.elsL10n` gets changed.

Comment (yzen) [inline apps/camera/js/views/controls.js]: There are a couple of things here, I think:
* l10n is not ready when the template is loaded and it does not get localized in time. localize for controls will be called on app localized event
* there is another PR on its way with localization for the labels for the rest of the controls, so I just wanted to keep it all together. I can do it without the loop as well.

File: apps/camera/js/controllers/preview-gallery.js (hunk 1 above)
Comment (justindarc) [inline apps/camera/js/controllers/preview-gallery.js]: We should have the view listen for the `wheel` DOM event from the screen reader and have the view emit a `swipe` event that this controller listens for. This controller is already handling `swipe` events that take a `direction` argument which is either 'left' or 'right'.

File: apps/camera/js/views/preview-gallery.js [hunk 3]
```diff
@@ -125,6 +130,17 @@ return View.extend({
...
   },
 
+  onFrameWheel: function(event) {
+    if (event.deltaMode !== event.DOM_DELTA_PAGE || event.deltaY) {
+      return;
+    }
+    if (event.deltaX > 0) {
+      this.fire('wheel:right');
+    } else if (event.deltaX < 0) {
+      this.fire('wheel:left');
```
Comment (justindarc) [inline apps/camera/js/views/preview-gallery.js]: This function should emit a `swipe` event instead. See below in `swipeCallback`:

```
self.emit('swipe', direction);
```

Where `direction` is either 'left' or 'right'.

File: apps/camera/js/views/controls.js (hunk 2 above)
Comment (justindarc) [inline apps/camera/js/views/controls.js]: No, my question was referring to the fact that I don't see where `this.elsL10n` gets localized values.

**Comment by try-server-hook** [comment]
[yzen Yura Zenevich (yzen)](https://github.com/yzen) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=53b81622f23a)

File: apps/camera/js/views/preview-gallery.js [hunk 4]
```diff
@@ -125,6 +130,17 @@ return View.extend({
     this.els.previewMenu.classList.remove('visible');
   },
 
+  onFrameWheel: function(event) {
+    if (event.deltaMode !== event.DOM_DELTA_PAGE || event.deltaY) {
+      return;
+    }
+    if (event.deltaX > 0) {
+      this.fire('swipe', 'left');
```
Comment (justindarc) [inline apps/camera/js/views/preview-gallery.js]: nit: I think we use `emit()` everywhere else, but `fire()` does the exact same thing

**Comment by try-server-hook** [comment]
[yzen Yura Zenevich (yzen)](https://github.com/yzen) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=e7166d8a7e40)

**Comment by try-server-hook** [comment]
[yzen Yura Zenevich (yzen)](https://github.com/yzen) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=45241dd66d7e)

**Pull request merged**